### PR TITLE
nfdump: add new package

### PR DESCRIPTION
**Pull request by flyn-org**
Signed-off-by: W. Michael Petullo <mike@flyn.org>

Maintainer: me
Compile tested: x86_64 master
Run tested: x86_64 master

Description:

### CODE REVIEW

File: net/nfdump/Makefile [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+PKG_INSTALL:=1
+PKG_BUILD_PARALLEL:=1
+
+include $(INCLUDE_DIR)/package.mk
+include $(INCLUDE_DIR)/nls.mk
+
+define Package/nfdump
+  SECTION:=net
+  CATEGORY:=Network
+  DEPENDS:= +libbz2 #+libpcap +libndpi +libcurl +libmaxminddb +libsqlite3 +libreadline +libopenssl +libzmq +librrd1 $(CXX_DEPENDS)
```
Comment (neheb) [inline net/nfdump/Makefile]: ?

File: net/nfdump/Makefile [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+PKG_NAME:=nfdump
+PKG_VERSION:=1.6.18
+PKG_RELEASE:=1
+
+PKG_MAINTAINER:=W. Michael Petullo <mike@flyn.org>
+
+PKG_LICENSE:=BSD-3-Clause
+PKG_LICENSE_FILES:=COPYING
+
+PKG_SOURCE:=v$(PKG_VERSION).tar.gz
```
Comment (neheb) [inline net/nfdump/Makefile]: Please no. v$(PKG_VERSION) is not a useful filename.

File: net/nfdump/Makefile [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+PKG_SOURCE_URL:=https://github.com/phaag/nfdump/archive
```
Comment (neheb) [inline net/nfdump/Makefile]: Use codeload instead.

File: net/nfdump/Makefile [hunk 4]
```diff
@@ -0,0 +1,64 @@
+#
+# Copyright (C) 2009-2012 OpenWrt.org
```
Comment (diizzyy) [inline net/nfdump/Makefile]: No, https://github.com/openwrt/packages/blob/master/CONTRIBUTING.md

File: net/nfdump/Makefile [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  TITLE:= nfdump
+  URL:=https://github.com/phaag/nfdump/
+endef
+
+define Package/nfdump/description
+	NetFlow collecting and processing tools
+endef
+
+define Build/Configure
+	( cd $(PKG_BUILD_DIR); ./autogen.sh )
```
Comment (diizzyy) [inline net/nfdump/Makefile]: Use PKG_FIXUP instead?

**Comment by flyn-org** [comment]
Force pushed modifications to address the comments from @diizzyy and @neheb.

File: net/nfdump/files/nfcapd.init [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+		echo "validation failed"
+		return 1
+	}
+
+	if [ "$enabled" -gt 0 ]; then
+		procd_open_instance
+		procd_set_param stdout 1
+		procd_set_param stderr 1
+		procd_set_param command "$PROG" -S 1 -P /var/run/nfcapd.pid -p "$port" -l "$logdir"
+		procd_close_instance
```
Comment (neheb) [inline net/nfdump/files/nfcapd.init]: I'd add

procd_set_param pidfile "$pidfile"

File: net/nfdump/files/nfcapd.init [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+		'logdir:string'
+}
+
+start_service() {
+	validate_nfcapd_section nfcapd || {
+		echo "validation failed"
+		return 1
+	}
+
+	if [ "$enabled" -gt 0 ]; then
```
Comment (neheb) [inline net/nfdump/files/nfcapd.init]: I'd do

[ "$enabled" -gt 0 ]; && { ... }

File: net/nfdump/files/nfcapd.init [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+		procd_open_instance
+		procd_set_param stdout 1
+		procd_set_param stderr 1
+		procd_set_param command "$PROG" -S 1 -P /var/run/nfcapd.pid -p "$port" -l "$logdir"
```
Comment (neheb) [inline net/nfdump/files/nfcapd.init]: I'd split into

procd_set_param command "$PROG"
procd_append_param command -S 1 -P /var/run/nfcapd.pid -p "$port" -l "$logdir"

File: net/nfdump/files/nfcapd.init (hunk 6 above)
Comment (neheb) [reply net/nfdump/files/nfcapd.init]: Actually, make sure to runtime test a change like that. I did with miniupnpd and it failed.

**Comment by flyn-org** [comment]
Pushed some changes. For some reason, 'procd_set_param pidfile "$pidfile"' prevented nfcapd from running, so I omitted it.

**Comment by flyn-org** [comment]
@neheb.

File: net/nfdump/Makefile [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+	NetFlow collecting and processing tools
+endef
+
+define Package/nfdump/conffiles
+/etc/config/nfcapd
+endef
+
+define Package/nfdump/install
+	$(INSTALL_DIR) $(1)/usr/lib
+	$(CP) $(PKG_INSTALL_DIR)/usr/lib/libnfdump-$(PKG_VERSION).so $(1)/usr/lib/
```
Comment (neheb) [inline net/nfdump/Makefile]: Is the so missing a * at the end?

File: net/nfdump/Makefile [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+
+	$(INSTALL_DIR) $(1)/usr/bin
+	$(CP) $(PKG_INSTALL_DIR)/usr/bin/* $(1)/usr/bin/
```
Comment (neheb) [inline net/nfdump/Makefile]: Should be INSTALL_BIN unless there are symlinks.

Comment (flyn-org) [reply net/nfdump/Makefile]: I updated the Makefile to use INSTALL_BIN.

File: net/nfdump/Makefile (hunk 9 above)
Comment (flyn-org) [reply net/nfdump/Makefile]: This is the only library file the binaries seem to expect.

**Comment by flyn-org** [comment]
@neheb, I updated the Makefile. See responses above for the details.

File: net/nfdump/Makefile (hunk 9 above)
Comment (neheb) [reply net/nfdump/Makefile]: Should be INSTALL_DATA then. .so files should not be executable.

Comment (flyn-org) [reply net/nfdump/Makefile]: Now uses INSTALL_DATA.

**Pull request merged**